efel/cppcore/Utils.cpp

```diff
@@ -200,6 +200,31 @@ std::pair<size_t, size_t> get_time_index(const std::vector<double>& t,
   return std::pair<size_t, size_t>(startIndex, endIndex);
 }
 
+// expects an empty newpeakindices
+int peaks_after_stim_start(const double stimstart,
+                           const vector<int>& peakindices,
+                           const vector<double>& t,
+                           vector<int>& newpeakindices){
+  for (size_t i=0; i < peakindices.size(); i++) {
+    if (t[peakindices[i]] > stimstart) {
+      newpeakindices.push_back(peakindices[i]);
+    }
+  }
+  return 1;
+}
+
+// expects an empty newpeakindices
+int peaks_after_stim_start(const double stimstartindex,
+                           const vector<int>& peakindices,
+                           vector<int>& newpeakindices){
+  for (size_t i=0; i < peakindices.size(); i++) {
+    if (peakindices[i] > stimstartindex) {
+      newpeakindices.push_back(peakindices[i]);
+    }
+  }
+  return 1;
+}
```

Comment:
Instead of the above function can't this one be used if `t[peakindices[i]]` is sent as an argument (instead of both t and peakindices)?

Reply:
I'm sorry, I'm not sure I understand. We don't have `i` yet when we call the function. Also the type would not match. Or am I missing something?

Reply:
Ah yes you're right. Ok then returning vector<int> directly here also would be more than enough :).

```
vector<int> peaks_after_stim_start(const double stimstartindex,
                                   const vector<int>& peakindices){
  vector<int> newpeakindices;
  for (size_t i = 0; i < peakindices.size(); i++) {
    if (peakindices[i] > stimstartindex) {
      newpeakindices.push_back(peakindices[i]);
    }
  }
  return newpeakindices;
}
```

Reply:
done in latest commit